test/test_ion_symbol.cpp

```diff
@@ -22,103 +22,93 @@
+    hREADER reader = NULL;
+    hWRITER writer = NULL;
+    ION_READER_OPTIONS reader_options = {0};
+    ION_WRITER_OPTIONS writer_options = {0};
+    ION_STREAM *stream = NULL;
 
-#define ION_SYMBOL_TEST_POPULATE_CATALOG \
-    ION_SYMBOL_TABLE *writer_imports[2]; \
-    hSYMTAB import1, import2; \
-    ION_STRING import1_name, import2_name, sym1, sym2, sym3; \
-    SID sid; \
-    hCATALOG catalog; \
-    ION_ASSERT_OK(ion_string_from_cstr("import1", &import1_name)); \
-    ION_ASSERT_OK(ion_string_from_cstr("import2", &import2_name)); \
-    ION_ASSERT_OK(ion_string_from_cstr("sym1", &sym1)); \
-    ION_ASSERT_OK(ion_string_from_cstr("sym2", &sym2)); \
-    ION_ASSERT_OK(ion_string_from_cstr("sym3", &sym3)); \
-    ION_ASSERT_OK(ion_catalog_open(&catalog)); \
-    ION_ASSERT_OK(ion_symbol_table_open_with_type(&writer_imports[0], catalog, ist_SHARED)); \
-    ION_ASSERT_OK(ion_symbol_table_open_with_type(&writer_imports[1], catalog, ist_SHARED)); \
-    import1 = writer_imports[0]; \
-    import2 = writer_imports[1]; \
-    ION_ASSERT_OK(ion_symbol_table_set_name(import1, &import1_name)); \
-    ION_ASSERT_OK(ion_symbol_table_set_name(import2, &import2_name)); \
-    ION_ASSERT_OK(ion_symbol_table_set_version(import1, 1)); \
-    ION_ASSERT_OK(ion_symbol_table_set_version(import2, 1)); \
-    ION_ASSERT_OK(ion_symbol_table_add_symbol(import1, &sym1, &sid)); \
-    ION_ASSERT_OK(ion_symbol_table_add_symbol(import2, &sym2, &sid)); \
-    ION_ASSERT_OK(ion_symbol_table_add_symbol(import2, &sym3, &sid)); \
-    ION_ASSERT_OK(ion_catalog_add_symbol_table(catalog, import1)); \
-    ION_ASSERT_OK(ion_catalog_add_symbol_table(catalog, import2));
+    ion_event_initialize_reader_options(&reader_options);
+    reader_options.pcatalog = catalog;
+    ION_ASSERT_OK(ion_reader_open_buffer(&reader, data, data_size, &reader_options));
+    ION_ASSERT_OK(ion_stream_open_memory_only(&stream));
 
-#define ION_SYMBOL_TEST_OPEN_WRITER_WITH_IMPORTS(is_binary, imports, import_count) \
-    ION_SYMBOL_TEST_DECLARE_WRITER; \
-    ION_WRITER_OPTIONS writer_options; \
-    ion_event_initialize_writer_options(&writer_options); \
-    ION_ASSERT_OK(ion_writer_options_initialize_shared_imports(&writer_options)); \
-    ION_ASSERT_OK(ion_writer_options_add_shared_imports_symbol_tables(&writer_options, imports, import_count)); \
-    writer_options.output_as_binary = is_binary; \
-    writer_options.pcatalog = catalog; \
-    ION_ASSERT_OK(ion_stream_open_memory_only(&stream)); \
+    writer_options.output_as_binary = as_binary;
+    ion_event_initialize_writer_options(&writer_options);
     ION_ASSERT_OK(ion_writer_open(&writer, stream, &writer_options));
+    ION_ASSERT_OK(ion_writer_write_all_values(writer, reader));
+    ION_ASSERT_OK(ion_reader_close(reader));
+    ION_ASSERT_OK(ion_test_writer_get_bytes(writer, stream, p_output, p_output_len));
+}
+
+void rewrite_and_assert_text_eq(BYTE *data, SIZE data_len, const char *expected) {
```

Comment:
PR Tour 🛫:  Provides functionality similar to the `ION_SYMBOL_TEST_REWRITE_FROM_WRITER_AND_ASSERT_TEXT` macro.